### PR TITLE
GCC 13 simulator build, console->simulator on Linux

### PR DESCRIPTION
**Pull request by nmschulte**
GCC 13 produces [warnings as] errors (for x86/simulator)

also make console search non `.exe` suffixed binaries, for e.g. Linux

### CODE REVIEW

File: firmware/hw_layer/drivers/can/can_hw.cpp [hunk 1]
```diff
@@ -32,10 +32,9 @@ class CanRead final : protected ThreadController<UTILITY_THREAD_STACK_SIZE> {
 	{
 	}
 
-	void start(CANDriver* device) {
-		m_device = device;
-
+	void tryStart(CANDriver *device) {
 		if (device) {
+			m_device = device;
```
Comment (mck1117) [inline firmware/hw_layer/drivers/can/can_hw.cpp:37]: why this change?

Comment (nmschulte) [reply firmware/hw_layer/drivers/can/can_hw.cpp:37]: Is unnecessary, leftover from when I was trying to properly override `start` (w/ separate `setDevice` method); can revert.

**Comment by nmschulte** [comment]
It works:

```
rlog Writing to logs/MAIN_rfi_report_2024-03-09_00_44_52_607.csv
I 240309 004452.615 [main] ConsoleUI - OS name: Linux
I 240309 004452.615 [main] ConsoleUI - OS version: 6.6.15-amd64
Console configuration from /home/nmschulte/.rusEFI/console_properties.xml
I 240309 004453.012 [AWT-EventQueue-0] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 0ms
I 240309 004453.082 [AWT-EventQueue-0] Launcher - Rendering available ports: []
I 240309 004453.321 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 0ms
I 240309 004453.626 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 1ms
I 240309 004453.930 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 0ms
I 240309 004454.234 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 0ms
I 240309 004454.539 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 1ms
I 240309 004454.843 [Ports Scanner] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 1ms
2024-03-09_00_44_54_939<EOT>: Executing ./fome_simulator
I 240309 004454.940 [AWT-EventQueue-0] TcpConnector - Connection refused in getAvailablePorts(): simulator not running in 0ms
rlog Writing to logs/SIMULATOR_CONSOLE_rfi_report_2024-03-09_00_44_54_940.csv
2024-03-09_00_44_54_951<EOT>: Executing ./fome_simulator=Process[pid=234531, exitValue="not exited"]
2024-03-09_00_44_56_941<EOT>: Port 29001 is alive
communicationThread lookup DONE
class com.rusefi.io.CommandQueue: SerialIO started
I 240309 004456.949 [AWT-EventQueue-0] ConsoleUI - init...
I 240309 004456.955 [AWT-EventQueue-0] ConsoleUI - FOME EFI Console 20231228
I 240309 004456.956 [AWT-EventQueue-0] LinkManager - LinkManager: Starting 29001
I 240309 004457.151 [AWT-EventQueue-0] ConsoleUI - showFrame
I 240309 004457.336 [AWT-EventQueue-0] Launcher - onWindowOpened
I 240309 004457.337 [AWT-EventQueue-0] StreamConnector - StreamConnector: connecting
I 240309 004457.338 [AWT-EventQueue-0] StreamConnector - scheduleOpening
I 240309 004457.338 [communication executor1] StreamConnector - scheduleOpening>openPort
class com.rusefi.io.LinkManager$4: Opening port: 29001
I 240309 004457.346 [[start] TCP reader1] ByteReader - [start] Running TCP connection loop
I 240309 004457.351 [communication executor1] BinaryProtocol - Got rusEFI (FOME) master.2024.03.09.f407-discovery.2031026796 signature
https://rusefi.com/online/ini/rusefi/(FOME) master/2024/03/09/f407-discovery/2031026796.ini
I 240309 004459.308 [communication executor1] BinaryProtocol - Reading from controller...
I 240309 004500.681 [communication executor1] BinaryProtocol - Got configuration from controller 23148 byte(s)
I 240309 004500.698 [AWT-EventQueue-0] BinaryProtocolServer - BinaryProtocolServer on 2390
I 240309 004500.700 [AWT-EventQueue-0] StatusConsumer - ServerSocket 2390 created. Feel free to point TS at IP Address 'localhost' port 2390
I 240309 004500.892 [communication executor1] MessagesCentral - postMessage EngineState: Length should be positive
I 240309 004500.892 [communication executor1] MessagesCentral - postMessage EngineState: Skipped zero-duration priming pulse.
I 240309 004500.893 [communication executor1] MessagesCentral - postMessage EngineState: Emulating position sensor(s). RPM=1200
I 240309 004500.893 [communication executor1] MessagesCentral - postMessage EngineState: LUA loading script length: 572...
I 240309 004500.893 [communication executor1] MessagesCentral - postMessage EngineState: LUA script loaded successfully!
I 240309 004500.893 [communication executor1] MessagesCentral - postMessage EngineState: Stimulator: updating trigger shape: 2/1
I 240309 004500.893 [communication executor1] MessagesCentral - postMessage EngineState: WARNING: out-of-order coil off Coil 1
I 240309 004500.895 [communication executor1] EngineSnifferPanel - Engine sniffer register channel i1 at idx -1
I 240309 004500.896 [communication executor1] EngineSnifferPanel - Engine sniffer register channel i2 at idx -1
I 240309 004500.896 [communication executor1] EngineSnifferPanel - Engine sniffer register channel i3 at idx -1
I 240309 004500.896 [communication executor1] EngineSnifferPanel - Engine sniffer register channel i4 at idx -1
I 240309 004500.896 [communication executor1] EngineSnifferPanel - Engine sniffer register channel map at idx -1
I 240309 004500.896 [communication executor1] EngineSnifferPanel - Engine sniffer register channel c1 at idx 1
I 240309 004500.901 [Commands Queue] BinaryProtocol - Sending [set date 2024-03-09T00:45:00.695366514]
```

File: java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java [hunk 2]
```diff
@@ -93,9 +94,25 @@ static void destroy() {
...
     }
+
+    private static File getSimulatorBinary() {
+        return getSimulatorBinary(SIMULATOR_BINARY_PATH + SIMULATOR_BINARY_NAME);
+    }
+
+    public static File getSimulatorBinary(String binaryPath) {
+        File binary = new File(binaryPath);
+
+        if (!binary.exists()) // try also for Windows/PE executable
```
Comment (mck1117) [inline java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java:110]: let's do { } instead of one line if

Comment (mck1117) [reply java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java:110]: (lest we `goto fail; goto fail;`)

File: java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java [hunk 3]
```diff
@@ -93,9 +94,27 @@ static void destroy() {
     }
 
     public static void startSimulator() {
-        if (!new File(SIMULATOR_BINARY).exists())
-            throw new IllegalStateException(SIMULATOR_BINARY + " not found");
+        getSimulatorBinary();
+
         FileLog.MAIN.logLine("startSimulator...");
         new Thread(SimulatorExecHelper::runSimulator, "simulator process").start();
```
Comment (mck1117) [inline java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java:100]: can we pass the result from line 97 in to the thread, instead of doing it again in the thread?

```java
final String b = getSimulatorBinary();
new Thread(() -> runSimulator(b)...)
```

File: java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java [hunk 4]
```diff
@@ -93,9 +94,27 @@ static void destroy() {
...
     }
+
+    private static File getSimulatorBinary() {
+        return getSimulatorBinary(SIMULATOR_BINARY_PATH + SIMULATOR_BINARY_NAME);
```
Comment (mck1117) [inline java_console/io/src/main/java/com/rusefi/SimulatorExecHelper.java:104]: is it worth requiring passing `SIMULATOR_BINARY_NAME` every time to `getSimulatorBinary`? Seems redundant.
